eng/Compilers.props

```diff
@@ -6,7 +6,18 @@
     <CompilerVersion>3.6.0-2.20166.2</CompilerVersion>
   </PropertyGroup>
 
+  <PropertyGroup Condition="!($(MSBuildProjectName.EndsWith('Tests', StringComparison.OrdinalIgnoreCase)))">
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    <NoWarn>$(NoWarn.Replace("1591", ""))</NoWarn> <!-- Make sure CS1591 is not in the list (by default it is) -->
+  </PropertyGroup>
+  <PropertyGroup Condition="($(MSBuildProjectName.EndsWith('Tests', StringComparison.OrdinalIgnoreCase)))">
+    <GenerateDocumentationFile>false</GenerateDocumentationFile>
+    <NoWarn>$(NoWarn)</NoWarn> <!-- No changes required here (For test projects, missing documentation is ok) -->
+  </PropertyGroup>
+  <PropertyGroup>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+  </PropertyGroup>
```

Comment:
How about changing this to one single property group to make it more readable like:

```suggestion
  <PropertyGroup>
    <_IsTestProject Condition="$(MSBuildProjectName.EndsWith('Tests', StringComparison.OrdinalIgnoreCase))">true</_IsTestProject>
    <NoWarn Condition="'$(_IsTestProject)' != 'true'">$(NoWarn.Replace("1591", ""))</NoWarn> <!-- Removing CS1591 from warning exclusion list as we want to make sure documentation is enforced -->
    <GenerateDocumentationFile Condition="'$(_IsTestProject)' != 'true'">true</GenerateDocumentationFile>
    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
  </PropertyGroup>
```

Reply:
@joperezr Thanks for sorting this out - I'm still learning the quirks of MsBuild projects and the property handling.